jscomp/syntax/ast_external_attributes.ml

```diff
@@ -128,8 +128,35 @@ let get_arg_type ~nolabel optional
        ptyp_desc = Ptyp_variant(List.rev rev_row_fields, Closed, None );
        ptyp_attributes
       }
-
     | (`Int, _), _ -> Bs_syntaxerr.err ptyp.ptyp_loc Invalid_bs_int_type
+    | (`Unwrap, ptyp_attributes), (Ptyp_variant (row_fields, Closed, _) as ptyp_desc) ->
+      let validity =
+        (List.fold_left
```

Comment:
can you make it a stand alone function and with some comments which are valid , which are invalid cases?